pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -99,6 +99,12 @@ func (admitter *VMsAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
+	// Skip any validation if we want to delete the vm
+	// This allows the vm controller to remove the finalizers
+	// also when, for example, the datasource is already gone.
+	if vm.DeletionTimestamp != nil && len(vm.Finalizers) == 0 {
+		return &admissionv1.AdmissionResponse{Allowed: true}
+	}
```

Comment:
This also assumes that there's only one finalizer relevant for VMs, as if we would have `anotherFinalizer` in the future, the length of `vm.Finalizers` could then be 1 therefore it'll prevent the controller from removing the controller finalizer if e.g. the data source doesn't exist.

In addition, this basically means that every VM update is legal when a VM is about to be deleted. Which validations are actually problematic here? If it's only the data source validation, then maybe we can define this as an exception especially for that case.

WDYT?

Reply:
> This also assumes that there's only one finalizer relevant for VMs, as if we would have `anotherFinalizer` in the future, the length of `vm.Finalizers` could then be 1 therefore it'll prevent the controller from removing the controller finalizer if e.g. the data source doesn't exist.

This is absolutely true! :)
 
> In addition, this basically means that every VM update is legal when a VM is about to be deleted. Which validations are actually problematic here? If it's only the data source validation, then maybe we can define this as an exception especially for that case.
> 
> WDYT?

This was exactly my doubt! I definitely encountered the problem with the dataSouce, but I suppose that there should be something else (? maybe?) that could cause the same issue! In my mind I though that, since this will be the last step before the vm disappear it should be fine!
But I am not sure, for this I want your feedback :)

I can also add some information: `sig-storage` lanes was always failing, so for sure the dataSource will always create the issue. But there was also a `sig-compute` failure once (https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8937/pull-kubevirt-e2e-k8s-1.23-sig-compute/1604848908272406528), but since there are no failing test (but only afterSuite), I am not able to find which one creates the pending VM.

Reply:
In general I think that seeing an error on a validation that's failing while the VMI is intended to be deleted is relatively easy to debug. A wrong and non-validated VM modification could lead to some weird side-effects that might be very hard to debug. So if we're talking only about the data source validation - I think it's better to move the `if` to there only. If many validations do not apply to a case of a VM that's being deleted and the whole validation isn't relevant that's a different story IMO.

@rmohr would appreciate your thoughts here as well

Reply:
Moved the check on ds